Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialFunctions/EvaluateEnhancedSurface.azsli

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
```

Comment:
My guess is that the correlation is quite high between the surfaces. Since it's quite a huge function, can we think of a way to have shared functionality that can be enhanced for the enhanced surface hence avoid repeating the same code and getting the advantage of maintaining less code differentiation?

Reply:
This was actually tried but it proved fairly difficult. The main goal of this change was really to define the interface, separating the material functions (which the material canvas will need to produce) and the implementation shaders (which are now SRG agnostic).